ddl/db_test.go

```diff
@@ -4213,6 +4213,26 @@ func (s *testDBSuite4) TestIssue9100(c *C) {
 	c.Assert(err.Error(), Equals, "[ddl:1503]A PRIMARY must include all columns in the table's partitioning function")
 }
 
+func (s *testSerialDBSuite) TestProcessColumnFlags(c *C) {
+	// check `processColumnFlags()`
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test_db")
+	tk.MustExec("create table t(a year(4) comment 'xxx', b year, c bit)")
+	defer s.mustExec(tk, c, "drop table t;")
+
+	tk.MustExec("alter table t modify a year(4)")
+
+	tk.MustExec("alter table t modify a year(4) signed")
+
+	tk.MustExec("alter table t modify a year(4) unsigned")
+
+	tk.MustExec("alter table t modify a year(4) zerofill")
+
+	tk.MustExec("alter table t modify b year")
+
+	tk.MustExec("alter table t modify c bit")
```

Comment:
Also check the new column type after altering.

Reply:
I only checked the flag, as this is what I really changed. Type change should have been tested somewhere else. And consistent type change is out of the scope of this PR.